### PR TITLE
[terraform-agent-2408, inf-terraform-[aws|azure]] feature/version bump 202408

### PR DESCRIPTION
**Pull request by tbugfinder**
- bump terraform, ruby versions
- bump AWS provider version, inspec-aws version
- bump pre-commit-hooks
- execute pre-commit-hooks and commit findings
- add jenkins-agent terraform-2408

Tasks: 
- [ ] Updated documentation in `docs/modules/...` directory
- [ ] Ran tests in `<quickstarter>/testdata` directory

### CODE REVIEW

**Comment by terra-naut** [comment]
Great job ❤️ 
Agent version needs to be updated to latest version (2408) https://github.com/opendevstack/ods-quickstarters/blob/master/inf-terraform-aws/Jenkinsfile.template
Azure too

File: common/jenkins-agents/terraform-2408/docker/python_requirements [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#  
+boto3~=1.34
+requests~=2.32
+simplejson~=3.19
+argparse~=1.4
+botocore
+pipenv~=2024.0
+python-hcl2~=2.0
+pre-commit~=3.7.1
```
Comment (terra-naut) [inline common/jenkins-agents/terraform-2408/docker/python_requirements:9]: pre-commit 3.8.0 ?

**Comment by tbugfinder** [comment]
> Great job ❤️ Agent version needs to be updated to latest version (2408) https://github.com/opendevstack/ods-quickstarters/blob/master/inf-terraform-aws/Jenkinsfile.template Azure too

thx, done

File: common/jenkins-agents/terraform-2408/docker/Dockerfile.ubi8 [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    CONSUL_VERSION=1.19.1 \
+    TENV_VERSION=3.0.0 \
+    TENV_AUTO_INSTALL=true \
+    TENV_ROOT=/opt/tenv \
+    GO_VERSION=1.21.13 \
+    GOBIN=/usr/local/go/bin \
+    TFLINT_VERSION=0.52.0 \
+    NODEJS_VERSION=20.16.0 \
+    BUNDLER_VERSION=2.5.17 \
+    SOPS_VERSION=3.9.0 \
```
Comment (braisvq1996) [inline common/jenkins-agents/terraform-2408/docker/Dockerfile.ubi8:29]: The installation of SOPS could be removed as it is already installed in the base agent -> https://github.com/opendevstack/ods-core/blob/master/jenkins/agent-base/Dockerfile.ubi8#L76

Comment (terra-naut) [reply common/jenkins-agents/terraform-2408/docker/Dockerfile.ubi8:29]: good catch - thx for the hint

**Comment by PfurtschellerP** [comment]
Hi @tbugfinder,
would it be possible to add the package `genisoimage` or a comparable one for iso image generation when running Packer against VMware infrastucture?

**Comment by tbugfinder** [comment]
> Hi @tbugfinder, would it be possible to add the package `genisoimage` or a comparable one for iso image generation when running Packer against VMware infrastucture?

Well, that's doable, would `xorriso` be okay, as this might be maintained compared to geniosimage?

File: common/jenkins-agents/terraform-2408/docker/Dockerfile.ubi8 (hunk 2 above)
Comment (tbugfinder) [reply common/jenkins-agents/terraform-2408/docker/Dockerfile.ubi8]: removed

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@
 - Update jdk and scala quickstarters and agents ([#1032](https://github.com/opendevstack/ods-quickstarters/issues/1032))
 - Gitleaks docs fix and update ([#1028](https://github.com/opendevstack/ods-quickstarters/issues/1028))
 - Enable OpenSSL vendored compilation for Rust Jenkins Agent ([#1026](https://github.com/opendevstack/ods-quickstarters/pull/1026))
+<<<<<<< HEAD
```
Comment (terra-naut) [inline CHANGELOG.md:13]: Looks like some merge leftovers

**Review by terra-naut — CHANGES_REQUESTED** [review]
Changelog needs to be fixed. Everything else : great Job

**Comment by terra-naut** [comment]
> > Hi @tbugfinder, would it be possible to add the package `genisoimage` or a comparable one for iso image generation when running Packer against VMware infrastucture?
> 
> Well, that's doable, would `xorriso` be okay, as this might be maintained compared to geniosimage?

Has that been resolved via a different channel? I see xorriso in the Dockerfile but no confirmation from @PfurtschellerP

**Comment by PfurtschellerP** [comment]
> > Hi @tbugfinder, would it be possible to add the package `genisoimage` or a comparable one for iso image generation when running Packer against VMware infrastucture?
> 
> Well, that's doable, would `xorriso` be okay, as this might be maintained compared to geniosimage?

Hi @tbugfinder,
yes that would also be fine.
Thank you!

**Review by terra-naut — APPROVED** [review]
Well done

**Review by terra-naut — APPROVED** [review]

File: inf-terraform-aws/Jenkinsfile.template [hunk 4]
```diff
@@ -3,7 +3,7 @@
 @Library('ods-jenkins-shared-library@@shared_library_ref@') _
 
 odsComponentPipeline(
-  imageStreamTag: '@ods_namespace@/jenkins-agent-terraform-2306:@agent_image_tag@',
```
Comment (BraisVQ) [inline inf-terraform-aws/Jenkinsfile.template:6]: This Quickstarter is managed by the Testing team, it is better if we do not modify it if not required and as it have not been updated to use the new agent we should left it as it is. I will inform them so they can check and update based on the updated AWS Quickstarter

**Review by BraisVQ — COMMENTED** [review]
I will do a quick tests to see if everything is ok, approve and merge

**Review by BraisVQ — APPROVED** [review]

**Pull request merged**